runtime/src/accounts_index.rs

```diff
@@ -1455,15 +1564,11 @@ impl<T: 'static + Clone + IsCached + ZeroLamport + std::marker::Sync + std::mark
             //  - The secondary index is never consulted as primary source of truth for gets/stores.
             //  So, what the accounts_index sees alone is sufficient as a source of truth for other non-scan
             //  account operations.
-            if let Some((mut w_account_entry, account_info)) =
-                self.get_account_write_entry_else_create(pubkey, slot, account_info)
-            {
-                w_account_entry.update(slot, account_info, reclaims);
-                false
-            } else {
-                true
-            }
-        };
+            let w_account_maps = self.get_account_maps_write_lock(pubkey);
```

Comment:
Primary concerns here are:

1) If a set of hotly updated pubkeys end up in the same bin, this could cause a performance degradation, especially if people are grinding for certain vanity pubkeys.

A good benchmark of this would be to generate a bunch of keys with the same first byte so they end up in the same bin, then run a set of transfers from those keys to another set of keys, and compare the perf to master.

2) RPC lookups (not even scans) for existing accounts like `get_balance` or `get_account` can no longer run concurrently with updates, so a lot of RPC lookups may starve progress here. This would be another good thing to try with an RPC node. I think Stephen has set up a bench for this before to test RPC scan + index contention, which we could repurpose for other RPC queries. 

The metric to look at for contention would be `accounts_db_store_timings::update_index`